elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/LabelStore.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.metadata;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.Path;
+import com.yahoo.elide.datastores.aggregation.core.JoinPath;
+
+import java.util.Set;
+
+/**
+ * LabelStore stores all label resolvers for all columns. It uses {@link JoinPath} as reference to each field.
+ */
+public interface LabelStore {
+    /**
+     * Get a dictionary with model definitions.
+     *
+     * @return dictionary
+     */
+    EntityDictionary getDictionary();
+
+    /**
+     * Get the label resolver for a path
+     *
+     * @param path path to a logical field
+     * @return a label resolver
+     */
+    LabelResolver getLabelResolver(Path path);
```

Comment:
Using Path here is unnatural.  If you look at the code, we only use the Class (Table) and field to actually store and lookup the resolvers.   There is a lot of code that has to lookup the resolver - and that code is creating these paths again and again.   Looking up by Path I think is confusing - because the actual lookup keys are obfuscated by the Path data structure.